Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -289,7 +289,35 @@ Further brute force mitigation techniques are described on the OWASP page [Block
 
 #### Dynamic Analysis
 
-Attempt to log in with an incorrect password multiple times. After three to five attempts, your login should be (at least temporarily) rejected even when correct credentials are entered.
+The purpose of dynamically analysing authentication mechanisms is to make sure an important number of login attempts cannot be made to guess credentials. 
+
+To test the level of confidence the authentication mechanism provides in an app, automation is often the preferred way. Many tools can be used, including scripts and proxies; the choice for the relevant tool may differ depending upon the situation. 
+For this demo, we'll use a proxy (Burp Suite Free edition). Burp Suite offers a module dedicated to such attacks called 'Intruder' (more information can be found from Burp Suite official documentation at [Burp Suite official documentation](https://portswigger.net/burp/help/ "Burp Suite official documentation")):
+
+  - start Burp Suite;
+  - create a new project (or open an existing one);
+  - enter relevant parameters for the project;
+  - now that the proxy is set between your app and the targeted web site, browse the site and intercept the request that corresponds to the authentication mechanism;
+  - in the 'Proxy / HTTP History' tab, right-click this request and select 'Send to Intruder' in the list; 
+  - select the 'Intruder' tab in Burp Suite;
+  - make sure all parameters in the 'Target', 'Positions' and 'Options' tabs are set accordingly, and select the 'Payload' tab;
+  - load the list of passwords to be tried or upload the list. The attack is ready to be started!
```

Comment:
Mention a number of passwords that should be used for this (maybe 25) and to add the correct password at the end of the list. If the login is still working after 25 failed login attempts, there is no security control in place to avoid brute force attacks.